seapath_ros_driver/include/seapath_ros_driver.hpp

```diff
@@ -0,0 +1,103 @@
+#ifndef SEAPATH_DRIVER_H
+#define SEAPATH_DRIVER_H
+
+#include <iostream>
+#include <sstream>
+#include <cmath> 
+#include <vector>
+#include <chrono>
+#include <memory>
+
+#include <rclcpp/rclcpp.hpp> 
+#include <sensor_msgs/msg/nav_sat_fix.hpp>
+#include <geometry_msgs/msg/twist_with_covariance_stamped.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <tf2/transform_datatypes.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+#include <vortex_msgs/msg/KMBinary.hpp>
+
+#include <seapath_socket.hpp>
+
+using std::placeholders::_1;
+using namespace std::chrono_literals;
+
+struct KMBinaryData {
+    char start_id[4];
+    uint16_t dgm_length;
+    uint16_t dgm_version;
+    uint32_t utc_seconds;
+    uint32_t utc_nanoseconds;
+    uint32_t status;
+    double latitude;
+    double longitude;
+    float ellipsoid_height;
+    float roll;
+    float pitch;
+    float heading;
+    float heave;
+    float roll_rate;
+    float pitch_rate;
+    float yaw_rate;
+    float north_velocity;
+    float east_velocity;
+    float down_velocity;
+    float latitude_error;
+    float longitude_error;
+    float height_error;
+    float roll_error;
+    float pitch_error;
+    float heading_error;
+    float heave_error;
+    float north_acceleration;
+    float east_acceleration;
+    float down_acceleration;
+    uint32_t delayed_heave_utc_seconds;
+    uint32_t delayed_heave_utc_nanoseconds;
+    float delayed_heave;
+};
+
+
+class SeaPathRosDriver : public rclcpp::Node{
+public: 
+    SeaPathRosDriver(const char* UDP_IP, const int UDP_PORT, std::chrono::duration<double> timerPeriod);
+    ~SeaPathRosDriver() = default;
+    KMBinaryData get_kmbinary_data();
+    void publish(KMBinaryData data);
+    geometry_msgs::msg::Point get_origin_publisher();
+
+private:
+    seapath_socket seaPathSocket;
+    KMBinaryData parse_kmbinary_data(std::vector<uint8_t> data);
+
+    geometry_msgs::msg::PoseWithCovarianceStamped to_pose_with_covariance_stamped(const KMBinaryData& data);
+    geometry_msgs::msg::TwistWithCovarianceStamped to_twist_with_covariance_stamped(const KMBinaryData& data);
+    
+    diagnostic_msgs::msg::DiagnosticStatus get_diagnostic_publisher();
+    sensor_msgs::msg::NavSatFix get_navsatfix_publisher(const KMBinaryData& data); 
+    vortex_msgs::msg::KMBinaryData kmbinary_publisher(const KMBinaryData& data);
```

Comment:
that is not the name of the vortex_msgs message, this should be "vortex_msgs::msg::KMBinary"